cognite/neat/graph/transformers/_value_type.py

```diff
@@ -0,0 +1,110 @@
+
+                            }}"""
+
+    _datatype_property_template: str = """SELECT ?s ?o WHERE {{
+
+                                ?s a <{subject_uri}> .
+                                ?s <{property_uri}> ?o .
+                                FILTER (datatype(?o) = <{datatype}>)
+
+                                }}"""
+
+    def transform(self, graph: Graph) -> None:
+        rules = self._infer_rules(graph)
+
+        class_reference_pairs = {class_.class_: class_.reference for class_ in rules.classes}
+
+        for property_ in InformationAnalysis(rules).multi_value_properties:
+            transformation = property_.transformation
+            if transformation and isinstance(transformation.traversal, SingleProperty):
+                traversal = transformation.traversal
+
+                for value_type in cast(MultiValueTypeInfo, property_.value_type).types:
+                    property_uri = rules.prefixes.get(traversal.property.prefix, rules.metadata.namespace)[
+                        traversal.property.suffix
+                    ]
+
+                    subject_uri = rules.prefixes.get(traversal.class_.prefix, rules.metadata.namespace)[
+                        traversal.class_.suffix
+                    ]
+
+                    # needs forming object_uri this way since we are also
+                    # translating original rdf types to rdf types in new
+                    # namespace so we need to pick the correct one which are
+                    # store in class references
+                    if isinstance(value_type, ClassEntity) and (
+                        object_uri := class_reference_pairs.get(value_type, None)
+                    ):
+                        for s, o in graph.query(  # type: ignore [misc]
+                            self._object_property_template.format(
+                                subject_uri=subject_uri,
+                                property_uri=property_uri,
+                                object_uri=object_uri,
+                            )
+                        ):
+                            graph.remove((s, property_uri, o))
+                            new_property = URIRef(f"{property_uri}_{value_type.suffix}")
+                            graph.add((s, new_property, o))
+
+                    elif isinstance(value_type, DataType) and not isinstance(value_type, AnyURI):
+                        for s, o in graph.query(  # type: ignore [misc]
+                            self._datatype_property_template.format(
+                                subject_uri=subject_uri,
+                                property_uri=property_uri,
+                                datatype=XSD[value_type.xsd],
+                            )
+                        ):
+                            graph.remove((s, property_uri, o))
+                            new_property = URIRef(f"{property_uri}_{value_type.xsd}")
+                            graph.add((s, new_property, o))
```

Comment:
Not sure if it is worth it, but warning of error there suddenly is something unexpected. 

The motivation for raising an error for a case that should not ever happen, is that in case it actually does you get a nice error message instead of something which is typically very hard to understand or you avoid silently failing. 

```suggestion
                            graph.add((s, new_property, o))
                    else:
                        raise ValueError(f"Value type {value_type} is not supported for splitting multi-value properties")
```